gprofiler/utils/node.py

```diff
@@ -0,0 +1,176 @@
+import json
+import os
+import shutil
+import signal
+import stat
+from typing import Dict, List
+
+import psutil
+import requests
+from granulate_utils.linux.ns import get_proc_root_path, get_process_nspid, resolve_proc_root_links, run_in_ns
+from granulate_utils.linux.process import is_musl
+from retry import retry
+from websocket import create_connection
+from websocket._core import WebSocket
+
+from gprofiler.log import get_logger_adapter
+from gprofiler.metadata import get_exe_version
+from gprofiler.utils import add_permission_dir, pgrep_exe, resource_path
+
+logger = get_logger_adapter(__name__)
+
+DSO_GIT_REVISION = "20eb88a"
```

Comment:
This requires manual updating which could be forgotten and then we cause harm by editing the file despite being loaded in some processes. We need to ensure gProfiler either uses the exact git revision OR refuses to use the file if the git revision is "incorrect".

I think it's easier to use the exact one - so like the AP build script, you can write the reivision to a file that is copied into our resources, and then add a function like `get_ap_version` which reads it?